src/smtlib_reader.cpp

```diff
@@ -50,6 +67,43 @@ int SmtLibReader::parse(const std::string & f)
 void SmtLibReader::set_logic(const string & logic)
 {
   solver_->set_logic(logic);
+
+  // process logic to get available operator symbols
+  string processed_logic = logic;
+  if (processed_logic.substr(0, 3) == "QF_")
+  {
+    // parser doesn't distinguish -- let solver complain
+    // if using quantifiers in quantifier-free logic
+    processed_logic = processed_logic.substr(3, processed_logic.length() - 3);
+  }
+
+  unordered_set<SortKind> sortkinds;
+  size_t logic_size;
+  while (logic_size = processed_logic.size())
+  {
+    // all existing theories have abbreviations of length 3 or shorter
+    for (size_t len = 1; len <= 3; len++)
```

Comment:
What about `NIRA` and `LIRA`? It would also be nice to support `QF_IDL` (integer difference logic) and `QF_RDL` (real difference logic). Both are a more restricted form of linear arithmetic logics.

Reply:
Addressed in 1d9005794323a62021ae90160404aa154cc58c53 and d4ba47ddb7a7a47c78a91712197d3bc767edd1f0